### PR TITLE
add easyblock for Molpro (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
Still testing this one, but early feedback can't hurt. :-)

**Comment by boegel** [comment]
@wpoely86: please review

File: easybuild/easyblocks/m/molpro.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                    raise EasyBuildError("Failed to create symlink for license token at %s", self.license_token)
+
+            else:
+                # no license token available, so fail early
+                raise EasyBuildError("No license token found at either %s or via 'license_file'", self.license_token)
+
+    def configure_step(self):
+        """Custom configuration procedure for Molpro: use 'configure -batch'."""
+
+        if not self.cfg['license_file'] or not os.path.exists(self.cfg['license_file']):
```
Comment (wpoely86) [inline easybuild/easyblocks/m/molpro.py]: Isn't this already sufficiently done in the `__init__`?

Comment (boegel) [inline easybuild/easyblocks/m/molpro.py]: Yeah, also, I had to move this to the `configure_step`, for compatibility with `--module-only`.

File: easybuild/easyblocks/m/molpro.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        for line in fileinput.input(cfgfile, inplace=1, backup='.orig'):
+            line = re.sub(r"^(LAUNCHER\s*=\s*).*$", r"\1 %s" % launcher, line)
+            sys.stdout.write(line)
+
+        # reread CONFIG and log contents
+        cfgtxt = read_file(cfgfile)
+        self.log.info("Contents of CONFIG file:\n%s", cfgtxt)
+
+        # make sure license token is available
+        token_path = os.path.join(os.environ['HOME'], '.molpro', 'token')
```
Comment (wpoely86) [inline easybuild/easyblocks/m/molpro.py]: why not do this at the beginning?

**Comment by wpoely86** [comment]
looks fine

File: easybuild/easyblocks/m/molpro.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/m/molpro.py]: ah, yes, this check can be removed, since we made sure it has been put in place at the start of `configure_step`

**Comment by boegel** [comment]
Tested with easyconfig in https://github.com/hpcugent/easybuild-easyconfigs/pull/1880, works as intended, so going in.

Thanks for the review @wpoely86!

**Pull request merged**